app/models/manageiq/providers/inventory/persister/builder/infra_manager.rb

```diff
@@ -275,13 +251,11 @@ def parent_blue_folders
             :custom_save_block => relationship_save_block(:relationship_key => :parent)
           )
 
-          add_dependency_attributes(
-            :ems_clusters   => ->(persister) { [persister.collections[:ems_clusters]] },
-            :ems_folders    => ->(persister) { [persister.collections[:ems_folders]] },
-            :hosts          => ->(persister) { [persister.collections[:hosts]] },
-            :resource_pools => ->(persister) { [persister.collections[:resource_pools]] },
-            :storages       => ->(persister) { [persister.collections[:storages]] },
-          )
+          dependency_collections = %i[ems_clusters ems_folders datacenters hosts resource_pools storages]
+          dependency_attributes = dependency_collections.each_with_object({}) do |collection, hash|
+            hash[collection] = ->(persister) { [persister.collections[collection]].compact }
```

Comment:
The `.compact` bit is the important part here since not all providers implement all of these collections.